test/Microsoft.ML.Tests/Scenarios/BinaryClassification.cs

```diff
@@ -0,0 +1,92 @@
+    {
+        public class BreastCancerData
+        {
+            [Column(ordinal: "0")]
+            public float Label;
+            [Column(ordinal: "1-9")]
+            [VectorType(9)]
+            public float[] Features;
+        }
+
+        public class BreastCancerPrediction
+        {
+            [ColumnName("PredictedLabel")]
+            public DvBool Cancerous;
+        }
+
+        [Fact]
+        public void SaveModelToOnnxTest()
+        {
+            string dataPath = GetDataPath(@"breast-cancer.txt");
+            var pipeline = new LearningPipeline();
+
+            pipeline.Add(new Data.TextLoader(dataPath)
+            {
+                Arguments = new TextLoaderArguments
+                {
+                    Separator = new[] { '\t' },
+                    HasHeader = true,
+                    Column = new[]
+                    {
+                        new TextLoaderColumn()
+                        {
+                            Name = "Label",
+                            Source = new [] { new TextLoaderRange(0) },
+                            Type = DataKind.Num
+                        },
+
+                        new TextLoaderColumn()
+                        {
+                            Name = "Features",
+                            Source = new [] { new TextLoaderRange(1, 9) },
+                            Type = DataKind.Num
+                        }
+                    }
+                }
+            });
+
+            pipeline.Add(new FastTreeBinaryClassifier() { NumLeaves = 5, NumTrees = 5, MinDocumentsInLeafs = 2 });
+            
+            PredictionModel<BreastCancerData, BreastCancerPrediction> model = pipeline.Train<BreastCancerData, BreastCancerPrediction>();
+            
+            var modelOutpath = GetOutputPath(Path.Combine("..", "Common", 
+                "Scenario", "BinaryClassification", "BreastCancer"), "SaveModelToOnnxTest.zip");
+
+            DeleteOutputPath(modelOutpath);
+
+            var onnxPath = GetOutputPath(Path.Combine("..", "Common",
+                "Scenario", "BinaryClassification", "BreastCancer"), "SaveModelToOnnxTest.pb");
+
+            DeleteOutputPath(onnxPath);
+
+            var onnxAsJsonPath = GetOutputPath(Path.Combine("..", "Common",
+                "Scenario", "BinaryClassification", "BreastCancer"), "SaveModelToOnnxTest.json");
+
+            DeleteOutputPath(onnxAsJsonPath);
+
+            model.WriteAsync(modelOutpath);
+            SaveAsOnnx.Save(new Runtime.Model.Onnx.SaveOnnxCommand.Arguments
+            {
+                InputModelFile = modelOutpath,
+                OutputsToDrop = "Label,Features",
+                Onnx = onnxPath,
+                Json = onnxAsJsonPath,
+                Domain = "Onnx"
+            });
+
+            Assert.True(CheckEquality(Path.Combine("..", "Common", "Scenario", "BinaryClassification", "BreastCancer"),
+                "SaveModelToOnnxTest.json"));
+        }
```

Comment:
If we're saying this is a baseline test, then it should behave like one. This means among other characteristics that it should avoid calling `Assert`s (which fill fail immediately), and instead let any failure happen in the final `Done()` call common to all baseline tests.

Reply:
I'm not actually certain you really want this to be like a baseline test though..

---
In reply to: [191591324](https://github.com/dotnet/machinelearning/pull/248#discussion_r191591324) [](ancestors = 191591324)

Reply:
I have removed the assert and added Done(). Why do you feel this shouldn't be a baseline test? I'm fine changing to anything else but this is a nice little example that shows the user how to covert the model to ONNX and in order for me to validate the model I need to compare the protobuf json output with a baseline. 

---
In reply to: [191591403](https://github.com/dotnet/machinelearning/pull/248#discussion_r191591403) [](ancestors = 191591403,191591324)

Reply:
*This* might be a baseline test, but your mechanism for making *it* a baseline test, is to make absolutely all tests be members of the baseline tests class ... e.g., `ScenariosTests : BaseTestBaseline`. I'm a little unenthusiastic about that.

---
In reply to: [191659361](https://github.com/dotnet/machinelearning/pull/248#discussion_r191659361) [](ancestors = 191659361,191591403,191591324)

Reply:
Ah ok, makes sense, I have removed this test from Scenario tests and made its own class called OnnxTests that extends BaseTestBaseline.

---
In reply to: [191829115](https://github.com/dotnet/machinelearning/pull/248#discussion_r191829115) [](ancestors = 191829115,191659361,191591403,191591324)

Reply:
Sounds like a good resolution, thanks Zeeshan.

---
In reply to: [192021512](https://github.com/dotnet/machinelearning/pull/248#discussion_r192021512) [](ancestors = 192021512,191829115,191659361,191591403,191591324)